network/Message.cpp

```diff
@@ -743,15 +825,33 @@ void ExtractErrorMessageData(const Message& msg, int& player_id, std::string& pr
     }
 }
 
-void ExtractHostMPGameMessageData(const Message& msg, std::string& host_player_name, std::string& client_version_string) {
+void ExtractHostMPGameMessageData(const Message& msg, std::string& host_player_name, std::string& client_version_string,
+                                  bool& use_compression) {
     try {
         std::istringstream is(msg.Text());
         freeorion_xml_iarchive ia(is);
         ia >> BOOST_SERIALIZATION_NVP(host_player_name)
-           >> BOOST_SERIALIZATION_NVP(client_version_string);
+           >> BOOST_SERIALIZATION_NVP(client_version_string)
+           >> BOOST_SERIALIZATION_NVP(use_compression);
+
+    } catch (const std::exception& err) {
+        ErrorLogger() << "ExtractHostMPGameMessageData(const Message& msg, std::string& host_player_name, std::string& client_version_string, bool& use_compression) failed!  Message:\n"
+                      << msg.Text() << "\n"
+                      << "Error: " << err.what();
+        throw err;
+    }
+}
+
+void ExtractHostMPAckMessage(const Message& msg, int& player_id, bool& req_compression)
```

Comment:
don't abbreviate to "req" as it could be "request" or "require"

Reply:
Function removed in https://github.com/freeorion/freeorion/pull/3854/commits/e937091c4540e765549e9147757117dfbe3f4bcf; I had previously added it so the removal is also a revert.